src/handlers/chain_agnostic/mod.rs

```
@@ -298,10 +298,14 @@ pub async fn get_assets_changes_from_simulation(
         if asset_changed.asset_type.clone() == AssetChangeType::Transfer
             && asset_changed.token_info.standard.clone() == TokenStandard::Erc20
             && asset_changed.to.is_some()
+            && asset_changed.token_info.contract_address.is_some()
         {
             asset_changes.push(Erc20AssetChange {
                 chain_id: transaction.chain_id.clone(),
-                asset_contract: asset_changed.token_info.contract_address,
+                asset_contract: asset_changed
+                    .token_info
+                    .contract_address
+                    .unwrap_or_default(),
```

We shouldn't use `unwrap_or_default()` here as it suppresses errors. Can we use `if let Some(..)`, or else offensive programming techniques?